### PR TITLE
ci: move to CircleCI

### PR DESCRIPTION
**Pull request by DeerMichel**
This PR removes the Travis CI pipeline in favor of CircleCI.

cc @zeke

### CODE REVIEW

**Comment by zeke** [comment]
@MarshallOfSound should we be seeing CIrcle CI checks showing up here?

**Comment by MarshallOfSound** [comment]
Build forked PR's was off by default, updated it now so it should build if you push again 👍

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+            - checkout
+            - restore_cache:
+                key: dependency-cache-{{ checksum "package.json" }}
+            - run: npm install
+            - save_cache:
+                key: dependency-cache-{{ checksum "package.json" }}
+                paths:
+                    - ./node_modules
+            - run: npm test
+            - run: npm run semantic-release
```
Comment (MarshallOfSound) [inline .circleci/config.yml]: We definitely do not want this in the `build` job.  Please check out an existing release config --> https://github.com/electron/asar/blob/master/.circleci/config.yml#L25-L64

File: package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
     "semantic-release": "semantic-release"
   },
   "devDependencies": {
+    "@continuous-auth/semantic-release-npm": "^1.0.3",
```
Comment (zeke) [inline package.json]: I'm unfamiliar with this dep. @DeerMichel do you have experience with this release setup and those plugins?

I usually use `npx semantic-release-cli setup` to configure this stuff, but if you're confident this setup will work for doing releases I'm okay with it.

Comment (DeerMichel) [reply package.json]: I guess @MarshallOfSound has 😄. All i did was following the [CFA guide](https://github.com/electron/cfa#set-up).

**Pull request merged**